### PR TITLE
Merge db into main

### PR DESCRIPTION
**Pull request by robynfsj**
DB complete. See notes document for all problems and future considerations. We might need to make modifications later, e.g. create an SQL function to calculate carbon saved or remove the function that calculates total emissions and write it in Python instead.

### CODE REVIEW

**Comment by robynfsj** [comment]
**NOTE:** I forgot to change branch from hw-project-brief branch to main before I created the create-db branch. The commits from the project description homework are on here but I removed the project description homework files later on. Can anyone help resolve this? I was hoping that everything would be fine as long as we merge the create-db branch before the hw-project-brief branch.

**Comment by rajwinderb** [comment]
SQL script worked for me, and all the tables were how I expected them to be! I tried it with Lakshika's code too and it works too.

**Review by rajwinderb — APPROVED** [review]
SQL script worked for me, and all the tables were how I expected them to be! I tried it with Lakshika's code too and it works too.

File: db/db-design-notes.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+- Column added for emissions saved.
+- The plan is that a user will enter origin, destination and vehicle. The total 
+  emissions for the journey with the specified vehicle is returned alongside a 
+  list of alternative vehicle options and their associated emissions. The user 
+  tells the app if they want to save their journey and which vehicle they ended 
+  up using. It is the data of the journey made and vehicle selected as the one 
+  they used that gets stored in the DB. The app also calculates the difference 
+  between their originally chosen vehicle and the vehicle they end up using and 
+  **stores this in the emissions saved column** for that journey.
+
```
Comment (Laksh-13) [inline db/db-design-notes.md:24]: So the function that we end up building for offset/saving needs to take into consideration emissions based on initial user choice and then emissions based on final user choice. The difference should result in the emissions saved.
I had initially envisaged it as user input to just be location and destination, modes of transport proposed by our service and offset calculated against the max emissions that could possibly be generated by a mode of transport in our proposed function. Of course the proposed mode of transport should ideally  never going to take into consideration any vehicle not listed in the user vehicle table.
Not sure if this makes sense?

File: db/db-design-notes.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+**TO DO**: Consider how carbon saved will be entered into table. Will it also 
+be done in SQL or will it be through a Python function when the user opts to 
+save their journey. Whichever we decide, it is probably best that both the 
+carbon emitted and carbon saved functions are in the same place/language.
+
```
Comment (Laksh-13) [inline db/db-design-notes.md:29]: I think, we should convert the emission calculation function into Python as well. 
We could potentially create a class that has methods that calculates emissions, offset etc. 
Would seem more standardized I guess ?

File: db/create-db.sql [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    DECLARE emitted INT;
+    
+    SELECT v.carb_emit_km INTO emissions_per_km
+    FROM vehicles AS v
+    WHERE v.vehicle_id = vehicle_id;
+    
+    SET emitted = distance * emissions_per_km;
+    RETURN emitted;
+END
+//
```
Comment (Laksh-13) [inline db/create-db.sql:126]: Migrate to Python maybe?

**Review by Laksh-13 — COMMENTED** [review]
Don't think we need to make any immediate changes...perhaps can remove the SQL emission accounting function later on :)

File: db/db-design-notes.md (hunk 1 above)
Comment (robynfsj) [reply db/db-design-notes.md:24]: Yes, I understand. This is just what I understood from our meeting the other day. We can do how you described here. Maybe we should have another proper discussion about it with everyone before we write the function.

**Review by Sravya12379 — APPROVED** [review]
Wow!! well done guys!! Everything looks perfect

**Review by clemcodes — APPROVED** [review]

**Pull request merged**